examples/seismic/acoustic/operators.py

```diff
@@ -206,30 +226,59 @@ def GradientOperator(model, geometry, space_order=4, save=True,
         Option to store the entire (unrolled) wavefield.
     kernel : str, optional
         Type of discretization, centered or shifted.
+    factor : int, optional
+        Downsampling factor to save snapshots of the wavefield.
     """
     m = model.m
 
-    # Gradient symbol and wavefield symbols
+    # Gradient symbol
     grad = Function(name='grad', grid=model.grid)
-    u = TimeFunction(name='u', grid=model.grid, save=geometry.nt if save
-                     else None, time_order=2, space_order=space_order)
-    v = TimeFunction(name='v', grid=model.grid, save=None,
-                     time_order=2, space_order=space_order)
-    rec = geometry.rec
+
+    # Create the adjoint wavefield
+    v = TimeFunction(name='v', grid=model.grid, time_order=2, space_order=space_order)
 
     s = model.grid.stepping_dim.spacing
     eqn = iso_stencil(v, model, kernel, forward=False)
 
-    if kernel == 'OT2':
-        gradient_update = Inc(grad, - u * v.dt2)
-    elif kernel == 'OT4':
-        gradient_update = Inc(grad, - u * v.dt2 - s**2 / 12.0 * u.biharmonic(m**(-2)) * v)
     # Add expression for receiver injection
+    rec = geometry.rec
     receivers = rec.inject(field=v.backward, expr=rec * s**2 / m)
 
+    time = model.grid.time_dim
+
+    if factor is not None:
+        # Condition to apply gradient update only at snapshot times
+        condition = Eq(time % factor, 0)
```

Comment:
No you don't need that `usnap` already contains the conditon

Reply:
Fixed.